test/integration/scheduler_perf/scheduler_perf_test.go

```diff
@@ -0,0 +1,273 @@
+		})
+	}
+}
+
+func perfScheduling(numNodes, numExistingPods, minPods int,
+	nodeStrategy testutils.PrepareNodeStrategy,
+	setupPodStrategy, testPodStrategy testutils.TestPodCreateStrategy,
+	nodeSpec *v1.Node, b *testing.B) {
+
+	finalFunc, podInformer, clientset := mustSetupScheduler()
+	defer finalFunc()
+
+	var nodePreparer testutils.TestNodePreparer
+	if nodeSpec != nil {
+		nodePreparer = framework.NewIntegrationTestNodePreparerWithNodeSpec(
+			clientset,
+			[]testutils.CountToStrategy{{Count: numNodes, Strategy: nodeStrategy}},
+			nodeSpec,
+		)
+	} else {
+		nodePreparer = framework.NewIntegrationTestNodePreparer(
+			clientset,
+			[]testutils.CountToStrategy{{Count: numNodes, Strategy: nodeStrategy}},
+			"scheduler-perf-",
+		)
+	}
+
+	if err := nodePreparer.PrepareNodes(); err != nil {
+		klog.Fatalf("%v", err)
+	}
+	defer nodePreparer.CleanupNodes()
+
+	config := testutils.NewTestPodCreatorConfig()
+	config.AddStrategy(setupNamespace, numExistingPods, setupPodStrategy)
+	podCreator := testutils.NewTestPodCreator(clientset, config)
+	podCreator.CreatePods()
+
+	for {
+		scheduled, err := getScheduledPods(podInformer)
+		if err != nil {
+			klog.Fatalf("%v", err)
+		}
+		if len(scheduled) >= numExistingPods {
+			break
+		}
+		klog.Infof("got %d existing pods, required: %d", len(scheduled), numExistingPods)
+		time.Sleep(1 * time.Second)
+	}
+
+	scheduled := int32(0)
+	completedCh := make(chan struct{})
+	podInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		UpdateFunc: func(old, cur interface{}) {
+			curPod := cur.(*v1.Pod)
+			oldPod := old.(*v1.Pod)
+
+			if len(oldPod.Spec.NodeName) == 0 && len(curPod.Spec.NodeName) > 0 {
+				if atomic.AddInt32(&scheduled, 1) >= int32(minPods) {
+					completedCh <- struct{}{}
+				}
+			}
+		},
+	})
+
+	// start benchmark
+	b.ResetTimer()
+	config = testutils.NewTestPodCreatorConfig()
+	config.AddStrategy(testNamespace, minPods, testPodStrategy)
+	podCreator = testutils.NewTestPodCreator(clientset, config)
+	podCreator.CreatePods()
+
+	<-completedCh
+
+	// Note: without this line we're taking the overhead of defer() into account.
+	b.StopTimer()
+}
+
+func getPodStrategy(pc podCase) testutils.TestPodCreateStrategy {
+	basePod := makeBasePod()
+	if pc.PodTemplatePath != nil {
+		basePod = getPodSpecFromFile(pc.PodTemplatePath)
+	}
+	if pc.ReplicationControllerName != nil {
+		return testutils.NewWithControllerCreatePodStrategy(*pc.ReplicationControllerName, basePod)
+	}
+	if pc.PersistentVolumeClaimTemplatePath == nil {
+		return testutils.NewCustomCreatePodStrategy(basePod)
+	}
+
+	pvTemplate := getPersistentVolumeSpecFromFile(pc.PersistentVolumeTemplatePath)
+	pvcTemplate := getPersistentVolumeClaimSpecFromFile(pc.PersistentVolumeClaimTemplatePath)
+	return testutils.NewCreatePodWithPersistentVolumeStrategy(pvcTemplate, getCustomVolumeFactory(pvTemplate), basePod)
+}
+
+func getTestCases(path string) []testCase {
+	bytes, err := ioutil.ReadFile(path)
```

Comment:
I think you can put the Unmarshall logic in a helper function (the `getSpec` function you had in initial version).